### ISSUE TITLE
Deprecation warning in sprockets 3.7.0

### ISSUE THREAD

**Issue by jaredbeck** [railsware/js-routes#201]
sprockets 3.7.0, js-routes 1.2.7, rails 5.0.0

```
DEPRECATION WARNING: You are using the a deprecated processor interface #<Proc:0x007fc451a92b68@/Users/jared/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/js-routes-1.2.7/lib/js_routes/engine.rb:11>.
Please update your processor interface:
https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors
 (called from block (2 levels) in <class:Engine> at /Users/jared/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/js-routes-1.2.7/lib/js_routes/engine.rb:11)
```

Looks like the [Extending Sprockets][1] guide is pretty thorough. I don't really *understand* any of it, but it looks pretty thorough :)

[1]: https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors

**Comment by jaredbeck** [comment]
Same warning in `sass-rails` (https://github.com/rails/sass-rails/issues/381) I suspect a lot of sprockets extensions will be fixing this in similar ways and may benefit from each other's work.

**Comment by bogdan** [comment]
Oh god...one more...

**Comment by jaredbeck** [comment]
> Same warning in sass-rails (rails/sass-rails#381) I suspect a lot of sprockets extensions will be fixing this in similar ways and may benefit from each other's work.

It's fixed in sass-rails, and according to https://github.com/rails/sass-rails/issues/381#issuecomment-234733694 it sounds like the sprockets documentation has been improved also, which should make this easier to fix here.

**Comment by WAFulbright** [comment]
nice commentaries.  HOW do I update this or fix it?

**Comment by flood4life** [comment]
I've fixed this in #202. It will still issue deprecation warning, but will work in sprockets 2, 3, 4. And I've found that guide to be not totally correct, but now it's fixed.

**Comment by bogdan** [comment]
Check the recent verison and ensure it fixes your issue.

**Comment by jaredbeck** [comment]
> Check the recent verison and ensure it fixes your issue.

Which version? The combination of `js-routes 1.2.8` and `sprockets 3.7.0` still produces this deprecation warning.

```
DEPRECATION WARNING: You are using the a deprecated processor interface #<Proc:0x007fac03b67680@/Users/jared/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/
gems/js-routes-1.2.8/lib/js_routes/engine.rb:58>.
Please update your processor interface:
https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md#supporting-all-versions-of-sprockets-in-processors
 (called from block (2 levels) in <class:Engine> at /Users/jared/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/
gems/js-routes-1.2.8/lib/js_routes/engine.rb:58)
```

```
bundle show sprockets
../gems/sprockets-3.7.0
bundle show js-routes
../gems/js-routes-1.2.8
```

**Comment by bogdan** [comment]
@flood4life can you take a look?

**Comment by flood4life** [comment]
@bogdan yep, exactly because of it, since you asked to not change anything for sprockets 2-3. Initial version (with single class) does not produce the warning. I think it's possible to explicitly suppress the warning, but not sure how.

**Comment by bogdan** [comment]
"Initial version" which one do you mean? 

Also, does it produce a warning for all 3.x versions or only for 3.7?

**Comment by flood4life** [comment]
The one before you asked to add the case statement. Not sure about versions (on mobile now), but here's the commit that adds deprecation https://github.com/rails/sprockets/commit/924c920648c163339abc6f3d790617d5be1b0bb0

**Comment by CyborgMaster** [comment]
I can confirm that the deprecation warning does not show up in 3.6.4, but does show up in 3.7.0.  I would love to get that warning out of my logs.

**Comment by bogdan** [comment]
We should change the condition to use newly developed model from #202 from `version >= 4.0` to `version >= 3.7`

**Comment by CyborgMaster** [comment]
Awesome.  Thanks!  I'll watch this issue here to see when I should update.

**Comment by bogdan** [comment]
Released 1.2.9. Please check

**Comment by jaredbeck** [comment]
Thanks Bogdan and Murad, I am no longer seeing this issue with js-routes 1.2.9. 👌
